wordpress-mobile/WordPress-Android
# Add Pin/Unpin functionality to dynamic cards

Fixes #13850 

This PR adds functionality to Pin and Unpin dynamic cards. I've introduced a new source that emits the order of the dynamic cards. This is then taken into account in the MySiteViewModel and reorders the dynamic cards. This is semi-future proof solution and I think it's quite simple and works well. 

To test:
- Make sure you have the MySiteImprovements flag turned on and the app is restarted
- Make sure quick start is in progress
- Click on the overflow menu on the bottom item
- Click on "Pin"
- Notice the item jumps to the top
- Restart the app
- Notice the pinned item is still on the top
- Click on the overflow menu
- Notice the button now says "Unpin"
- Click on "Unpin"
- Notice the card now jumps to the bottom

https://user-images.githubusercontent.com/1079756/107228868-6e2d7200-6a1d-11eb-9614-ea925ec50318.mp4


PR submission checklist:
